pkg/buildpack/build_module_info.go

```diff
@@ -0,0 +1,65 @@
+package buildpack
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/pkg/dist"
+)
+
+type ModuleInfos interface {
+	BuildModule() []dist.ModuleInfo
+}
+
+type FlattenModuleInfos interface {
+	FlattenModules() []ModuleInfos
+}
+
+type flattenModules struct {
+	modules []ModuleInfos
+}
+
+func (fl *flattenModules) FlattenModules() []ModuleInfos {
+	return fl.modules
+}
+
+type buildModuleInfosImpl struct {
+	modules []dist.ModuleInfo
+}
+
+func (b *buildModuleInfosImpl) BuildModule() []dist.ModuleInfo {
+	return b.modules
+}
+
+func ParseFlattenBuildModules(buildpacksID []string) (FlattenModuleInfos, error) {
```

Comment:
Could you add some unit tests for this one?

Reply:
Yeah, I have to do it! thanks for the feedback ❤️